numpy/fft/helper.py

```diff
@@ -216,7 +227,11 @@ def rfftfreq(n, d=1.0):
     """
     if not isinstance(n, integer_types):
         raise ValueError("n should be an integer")
-    val = 1.0/(n*d)
+    if dtype != float32:
+        dtype = float64
+
+    val = reciprocal(n*d, dtype=dtype)
     N = n//2 + 1
     results = arange(0, N, dtype=int)
```

Comment:
Sorry one last thought. Would it make sense to cast this to `dtype` as well?

```suggestion
    results = arange(0, N, dtype=dtype)
```

Reply:
we could but from what I understand this is always just an array of integers, which when multiplied with the `dtype` cast `val` will lead to the result being of type `dtype` anyway, right?